src/main/java/at/hannibal2/skyhanni/features/itemabilities/abilitycooldown/ItemAbilityCooldown.kt

```diff
@@ -428,13 +455,26 @@ object ItemAbilityCooldown {
         val itemName: String = stack.cleanName()
         val internalName = stack.getInternalName()
 
+
         val list = mutableListOf<ItemAbility>()
         for (ability in ItemAbility.entries) {
             if (ability.newVariant) {
                 if (ability.internalNames.contains(internalName)) {
                     list.add(ability)
                 }
-            } else {
+                for (scroll in stack.getAbilityScrolls()?.toList().orEmpty())
+                {
+                    if (ability.internalNames.contains(scroll)){
+                    list.add(ability)
+                    }
+                }
+                if (list.containsAll((listOf(WITHER_SHIELD_SCROLL, IMPLOSION_SCROLL, SHADOW_WARP_SCROLL))))
```

Comment:
checking `containsAll` is unnecessary. The list can only have 0-3 entries. So if there are 3 entries, it is those three and we don't need to check fi it is actually those three.